code/surface_area.py

```diff
@@ -0,0 +1,421 @@
+import argparse
+from pathlib import Path
+from typing import Iterable
+import numpy as np
+
+from niwrap import workbench as wb
+import nibabel as nib
+
+
+def detect_density(n_vertices: int) -> str:
+    """
+    Map a vertex count to a density label for (NHP) surface meshes.
+    Returns a string label like '32k', '100k' or the exact integer as string for very large/unknown meshes.
+    """
+    if n_vertices < 3000:
+        return "2k"
+    if 3000 <= n_vertices <= 7499:
+        return "5k"
+    if 7500 <= n_vertices <= 12499:
+        return "10k"
+    if 12500 <= n_vertices <= 17499:
+        return "15k"
+    if 17500 <= n_vertices <= 22499:
+        return "20k"
+    if 22500 <= n_vertices <= 27499:
+        return "25k"
+    if 27500 <= n_vertices <= 36499:
+        return "32k"
+    if 36500 <= n_vertices <= 45999:
+        # many NHP "32k" meshes sit here (e.g. 40962). Use 40k label to be explicit,
+        # but you can override with a template lookup if you want HCP-style naming.
+        return "40k"
+    if 46000 <= n_vertices <= 55999:
+        return "50k"
+    if 56000 <= n_vertices <= 70999:
+        return "59k"
+    if 71000 <= n_vertices <= 95999:
+        return "80k"
+    if 96000 <= n_vertices <= 120999:
+        return "100k"
+    if 121000 <= n_vertices <= 140999:
+        return "120k"
+    if 141000 <= n_vertices <= 180999:
+        return "164k"
+    # fallback: return exact count for uncommon/custom meshes
+    return str(n_vertices)
+
+
+# Map hemisphere from structure name
+def detect_hemi(structure_name: str, filename: str) -> str:
+    struct_map = {
+        "CortexLeft": "L",
+        "CortexRight": "R"
+    }
+    if structure_name in struct_map:
+        return struct_map[structure_name]
+    # fallback to filename patterns
+    lower = filename.lower()
+    if "lh" in lower or "left" in lower or ".l." in lower:
+        return "L"
+    if "rh" in lower or "right" in lower or ".r." in lower:
+        return "R"
+    return "na"
+
+
+# Detect space from filename or folder
+def detect_space(path: Path) -> str:
+    candidates = ["D99", "MEBRAINS", "Yerkes19", "CIVETNMT"]
+    for c in candidates:
+        if c.lower() in path.lower():
+            return c
+    return "unknown"
```

Comment:
I would consider throwing an exception or exiting here if it is unknown since we should know what templates to expect..